workos/resources/webhooks.py

```diff
@@ -0,0 +1,283 @@
+from typing import Generic, Literal, Union
+from pydantic import Field
+from typing_extensions import Annotated
+from workos.resources.directory_sync import DirectoryGroup
+from workos.resources.events import EventPayload
+from workos.resources.user_management import OrganizationMembership, User
+from workos.resources.workos_model import WorkOSModel
+from workos.types.directory_sync.directory_user import DirectoryUser
+from workos.types.events.authentication_payload import (
+    AuthenticationEmailVerificationSucceededPayload,
+    AuthenticationMagicAuthFailedPayload,
+    AuthenticationMagicAuthSucceededPayload,
+    AuthenticationMfaSucceededPayload,
+    AuthenticationOauthSucceededPayload,
+    AuthenticationPasswordFailedPayload,
+    AuthenticationPasswordSucceededPayload,
+    AuthenticationSsoSucceededPayload,
+)
+from workos.types.events.connection_payload_with_legacy_fields import (
+    ConnectionPayloadWithLegacyFields,
+)
+from workos.types.events.directory_group_membership_payload import (
+    DirectoryGroupMembershipPayload,
+)
+from workos.types.events.directory_group_with_previous_attributes import (
+    DirectoryGroupWithPreviousAttributes,
+)
+from workos.types.events.directory_payload import DirectoryPayload
+from workos.types.events.directory_payload_with_legacy_fields import (
+    DirectoryPayloadWithLegacyFields,
+)
+from workos.types.events.directory_user_with_previous_attributes import (
+    DirectoryUserWithPreviousAttributes,
+)
+from workos.types.events.organization_domain_verification_failed_payload import (
+    OrganizationDomainVerificationFailedPayload,
+)
+from workos.types.events.session_created_payload import SessionCreatedPayload
+from workos.types.organizations.organization_common import OrganizationCommon
+from workos.types.organizations.organization_domain import OrganizationDomain
+from workos.types.roles.role import Role
+from workos.types.sso.connection import Connection
+from workos.types.user_management.email_verification_common import (
+    EmailVerificationCommon,
+)
+from workos.types.user_management.invitation_common import InvitationCommon
+from workos.types.user_management.magic_auth_common import MagicAuthCommon
+from workos.types.user_management.password_reset_common import PasswordResetCommon
+
+
+class WebhookModel(WorkOSModel, Generic[EventPayload]):
+    """Representation of an Webhook delivered via Webhook.
+    Attributes:
+        OBJECT_FIELDS (list): List of fields an Webhook is comprised of.
+    """
+
+    id: str
+    data: EventPayload
+    created_at: str
```

Comment:
Super-sad panda 😿 

There is a difference between our webhooks and events. Each event contains `object: 'event'`. Our webhooks do not.

A mini future us project should be to add this `object` attributes to webhook bodies so that they're identical and we can remove this sort of duplication.